pkg/ottl/expression.go

```diff
@@ -360,7 +401,7 @@ func (g StandardPMapGetter[K]) Get(ctx context.Context, tCtx K) (pcommon.Map, er
 		if v.Type() == pcommon.ValueTypeMap {
 			return v.Map(), nil
 		}
-		return pcommon.Map{}, TypeError(fmt.Sprintf("expected pcommon.Map but got %v", v.Type()))
+		return pcommon.Map{}, TypeError(fmt.Sprintf("expected pcommon.Map but got %T", v))
```

Comment:
```suggestion
		return pcommon.Map{}, TypeError(fmt.Sprintf("expected pcommon.Map but got %v", v.Type()))
```
Sorry, I got this wrong before. Since this is a pdata object, we should use its `Type` method.